cmake/modules/AddPureSwift.cmake

```diff
@@ -1,11 +1,8 @@
 include(macCatalystUtils)
 
 # Workaround a cmake bug, see the corresponding function in swift-syntax
-function(force_target_link_libraries TARGET)
-  target_link_libraries(${TARGET} ${ARGN})
-
-  cmake_parse_arguments(ARGS "PUBLIC;PRIVATE;INTERFACE" "" "" ${ARGN})
-  foreach(DEPENDENCY ${ARGS_UNPARSED_ARGUMENTS})
+function(force_add_dependencies TARGET)
+  foreach(DEPENDENCY ${ARGN})
```

Comment:
Split `force_target_link_libraries` so that we can just add dependencies without linking.